In [ ]:
!pip install -q xlrd
!git clone https://github.com/biacraciun/Understanding-Context-Usage-in-MT

In [ ]:
!pip install -q -r Understanding-Context-Usage-in-MT/requirements.txt

In [ ]:
# Restart the session before executing this block of code so that the dependencies are loaded properly

import sys

# Add the path to the folder containing the modules
sys.path.append('Understanding-Context-Usage-in-MT/modules')

import setup_data_pecore
import use_pecore
import run_pecore

In [ ]:
from run_pecore import run_pecore

In [ ]:
run_pecore(langugage_code='ron_Latn', notebook=True)

config.json:   0%|          | 0.00/846 [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/2.46G [00:00<?, ?B/s]

generation_config.json:   0%|          | 0.00/189 [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/564 [00:00<?, ?B/s]

sentencepiece.bpe.model:   0%|          | 0.00/4.85M [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/17.3M [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/3.55k [00:00<?, ?B/s]

Sentence 12


Context with contextual cues (std λ=2.00, top 5) followed by output sentence with context-sensitive target spans 
(all)
(CTI = "kl_divergence", CCI = "saliency" w/ "probability" target)

Input context:  Ecuador requires that Cuban citizens receive an invitation letter prior to entering Ecuador through
international airports or frontier admission points. This letter must be legalized by the Ecuadorian Ministry of 
Foreign Affairs, and comply with certain requirements. These requirements are designed to provide an organized 
migratory flow between both countries. Cuban citizens who are US green card holders should visit an Ecuadorian 
Consulate to obtain an exemption to this requirement.
Input current: Your passport must be valid for at least 6 months beyond your travel dates. A round/onward trip 
ticket is needed to prove the length of your stay.
Output context: Ecuadorul cere ca cetățenii Cubei să primească o scrisoare de invitație înainte de a intra în 
Ecuador folosind aeroporturile internaționale sau punctele de trecere a frontierei. Scrisoarea aceasta trebuie să 
fie legalizată de către Ministerul Afacerilor Externe al Ecuadorului și trebuie să respecte anumite cerințe. 
Cerințele acestea sunt concepute pentru a asigura o deplasare bine organizată între cele două țări. Cetățenii 
cubanezi care dețin cărți verzi pentru SUA trebuie să viziteze un consulat ecuadorian pentru a obține o scutire de 
la această cerință.
Output current: Pașaportul dumneavoastră trebuie să fie valabil pentru cel puțin 6 luni după datele de călătorie. 
Un bilet de călătorie în direcție / în continuare este necesar pentru a dovedi durata șederii dumneavoastră.

#1.
Generated output (CTI > 0.165):  Pa(1.870)șaportul dumneavoastră trebuie să fie valabil pentru cel puțin 6 luni 
după datele de călătorie. Un bilet de călătorie în direcție / în continuare este necesar pentru a dovedi durata 
șederii dumneavoastră.
Input context (CCI > 0.001):     Ecuador requires that Cuban citizens receive an invitation letter prior to 
entering Ecuador through international airports or frontier admission points. This letter must be legalized by 
the(0.002) Ecuadorian Ministry of Foreign Affairs, and comply with certain requirements. These requirements are 
designed to provide an organized migratory flow between both countries. Cuban citizens who are US green card 
holders should visit an(0.001) Ecuadorian Consulate to obtain an exemption to this requirement(0.003).(0.002)
Output context (CCI > 0.001):    Ecuadorul cere ca cetățenii Cubei să primească o scrisoare de invitație înainte de
a intra în Ecuador folosind aeroporturile internaționale sau punctele de trecere a frontierei. Scrisoarea aceasta 
trebuie să fie legalizată de către Ministerul Afacerilor Externe al Ecuadorului și trebuie să respecte anumite 
cerințe. Cerințele acestea sunt concepute pentru a asigura o deplasare bine organizată între cele două țări. 
Cetățenii cubanezi care dețin cărți verzi pentru SUA trebuie să viziteze un consulat ecuadorian pentru a obține o 
scutire de la această cerință.(0.006)
#2.
Generated output (CTI > 0.165):  Pașaportul dumneavoastră trebuie să fie valabil pentru cel puțin 6 luni după 
datele de călătorie. Un(1.225) bilet de călătorie în direcție / în continuare este necesar pentru a dovedi durata 
șederii dumneavoastră.
Input context (CCI > 0.001):     Ecuador requires that Cuban citizens receive an invitation letter prior to 
entering Ecuador through international airports or frontier admission points. This letter must be legalized by 
the(0.001) Ecuadorian Ministry of Foreign Affairs, and comply with certain requirements. These requirements are 
designed to provide an organized migratory flow between both countries. Cuban citizens who are US green card 
holders should visit an(0.001) Ecuadorian Consulate to obtain an exemption to this requirement(0.001).
Output context (CCI > 0.001):    Ecuadorul cere ca cetățenii Cubei să primească o scrisoare de invitație înainte de
a int

Sentence 18


Context with contextual cues (std λ=2.00, top 5) followed by output sentence with context-sensitive target spans 
(all)
(CTI = "kl_divergence", CCI = "saliency" w/ "probability" target)

Input context:  There are many things you have to take into consideration before and when you travel somewhere.
Input current: When you travel, expect things not to be like they are "back home". Manners, laws, food, traffic, 
lodging, standards, language and so on will to some extent differ from where you live.
Output context: Sunt multe lucruri de luat în considerare înainte și în timpul călătoriei spre o anumită 
destinație.
Output current: Când călătorești, așteaptă să nu fie ca "în casă".

#1.
Generated output (CTI > 0.124):  Când(0.901) călătorești, așteaptă să nu fie ca "în casă".
Input context (CCI > 0.010):     There are many things you have to take into consideration before and when you 
travel somewhere(0.011).
Output context (CCI > 0.011):    Sunt multe lucruri de luat în considerare înainte și în timpul călătoriei spre o 
anumită destinație.(0.024)
#2.
Generated output (CTI > 0.124):  Când călătorești(0.667), așteaptă să nu fie ca "în casă".
Input context (CCI > 0.007):     There are many things you have to take into consideration before and when you 
travel somewhere.
Output context (CCI > 0.008):    Sunt multe lucruri de luat în considerare înainte și(0.015) în timpul călătoriei 
spre o anumită destinație.

Sentence 25


Context with contextual cues (std λ=2.00, top 5) followed by output sentence with context-sensitive target spans 
(all)
(CTI = "kl_divergence", CCI = "saliency" w/ "probability" target)

Input context:  At home, due to this constant exposure to the local germs, odds are very high that you're already 
immune to them.
Input current: But in other parts of the world, where the bacteriological fauna are new to you, you're far more 
likely to run into problems.
Output context: Acasă, datorită expunerii constante la microbii locali, șansele de a fi deja imun la ei sunt foarte
mari.
Output current: Dar în alte părți ale lumii, unde fauna bacteriologică este nouă pentru tine, e mult mai probabil 
să ai probleme.

#1.
Generated output (CTI > 0.045):  Dar în alte părți(0.534) ale lumii, unde fauna bacteriologică este nouă pentru 
tine, e mult mai probabil să ai probleme.
Input context (CCI > 0.003):     At home, due to this constant exposure to the local germs, odds are very high that
you're already immune to them.
Output context (CCI > 0.003):    Acasă, datorită expunerii constante la microbii locali, ș(0.009)ansele de a fi 
deja imun la ei sunt foarte mari.
#2.
Generated output (CTI > 0.045):  Dar(0.389) în alte părți ale lumii, unde fauna bacteriologică este nouă pentru 
tine, e mult mai probabil să ai probleme.
Input context (CCI > 0.004):     At home, due to this constant exposure to the local germs, odds are very high that
you're already immune to them(0.005).
Output context (CCI > 0.004):    Acasă, datorită expunerii constante la microbii locali, șansele de a fi deja imun 
la ei sunt foarte mari.(0.009)
#3.
Generated output (CTI > 0.045):  Dar în alte părți ale lumii, unde fauna bacteriologică este nouă pentru tine, 
e(0.057) mult mai probabil să ai probleme.
Input context (CCI > 0.001):     At home, due to this constant exposure to the local germs, odds are very high that
you're already immune to them.
Output context (CCI > 0.001):    Acasă, datorită expunerii constante la microbii locali, ș(0.002)ansele de a fi 
deja imun la ei sunt foarte mari.

Sentence 29


Context with contextual cues (std λ=2.00, top 5) followed by output sentence with context-sensitive target spans 
(all)
(CTI = "kl_divergence", CCI = "saliency" w/ "probability" target)

Input context:  Fill your home with a rich coffee in the morning and some relaxing chamomile tea at night. When 
you're on a staycation, you have the time to treat yourself and take a few extra minutes to brew up something 
special. If you're feeling more adventurous, take the opportunity to juice or blend up some smoothies:
Input current: maybe you'll discover a simple beverage that you can make for breakfast when you're back to your 
daily routine.
Output context: Umple-ți casa cu o aromă de cafea savuroasă dimineața și de ceai de mușețel relaxant seara. Când 
stați consemnat la domiciliu, aveți timp să vă faceți plăcerile și să vă permiteți câteva minute extra ca să 
pregătiți o masă specială. Dacă ai chef de aventură, profită de ocazie ca să faci un suc sau să amesteci niște 
smoothies:
Output current: poate vei descoperi o băutură simplă pe care o poți face pentru micul dejun când te-ai întors la 
rutina de zi cu zi.

#1.
Generated output (CTI > 0.161):  poate(2.328) vei descoperi o băutură simplă pe care o poți face pentru micul dejun
când te-ai întors la rutina de zi cu zi.
Input context (CCI > 0.001):     Fill your home with a rich coffee in the morning and some relaxing chamomile tea 
at night. When you're on a staycation, you have the time to treat yourself and take a few extra minutes to brew up 
something special. If you're feeling more adventurous, take the opportunity to juice or blend up some 
smoothies(0.003):(0.003)
Output context (CCI > 0.001):    Umple-ți casa cu o aromă de cafea savuroasă dimineața și de ceai de mușețel 
relaxant seara. Când stați consemnat la domiciliu, aveți timp să vă faceți plăcerile și să vă permiteți câteva 
minute extra ca să pregătiți o masă specială. Dacă ai chef de aventură, profită de ocazie ca să faci un suc sau să 
amesteci niște smoothies:(0.002)
#2.
Generated output (CTI > 0.161):  poate vei descoperi o băutură simplă pe care o po(1.392)ți face pentru micul dejun
când te-ai întors la rutina de zi cu zi.
Input context (CCI > 0.002):     Fill your home with a rich coffee in the morning and some relaxing chamomile tea 
at night. When you're on a staycation, you have the time to treat yourself and take a few extra minutes to brew up 
something special. If you're feeling more adventurous, take the opportunity to juice or blend up some smoothies:
Output context (CCI > 0.002):    Umple-ți casa cu o aromă de cafea savuroasă dimineața și de ceai de mușețel 
relaxant seara. Când stați(0.003) consemnat la domiciliu, ave(0.003)ți(0.005) timp să vă faceți plăcerile și să vă 
permiteți câteva minute extra ca să pregătiți o masă specială. Dacă ai chef de aventură, profită de ocazie ca să 
faci un suc sau să am(0.004)este(0.005)ci niște smoothies:
#3.
Generated output (CTI > 0.161):  poate vei descoperi o băutură simplă pe care o poți face pentru micul dejun când 
te-(0.355)ai întors la rutina de zi cu zi.
Input context (CCI > 0.002):     Fill your home with a rich coffee in the morning and some relaxing chamomile tea 
at night. When you're on a staycation, you have the time to treat yourself and take a few extra minutes to brew up 
something special. If you're feeling more adventurous, take the opportunity to juice or blend up some smoothies:
Output context (CCI > 0.002):    Umple-ți casa cu o aromă de(0.004) cafea savuroasă dimineața și(0.002) de ceai de 
mușețel relaxant seara. Când stați consemnat la domiciliu, aveți timp să vă faceți plăcerile și să vă permiteți 
câteva minute extra ca să pregătiți o masă specială. Dacă ai chef de(0.005) aventură, profită de ocazie ca să faci 
un suc sau să amesteci ni(0.003)ște smoothies:
#4.
Generated output (CTI > 0.161):  poate vei(0.332) descoperi o băutură simplă pe care o poți face pentru micul dejun
când te-ai întors la rutina de zi cu zi.
Input context (CCI > 0

Sentence 39


Context with contextual cues (std λ=2.00, top 5) followed by output sentence with context-sensitive target spans 
(all)
(CTI = "kl_divergence", CCI = "saliency" w/ "probability" target)

Input context:  In summer, especially, you'll need to watch out for mosquitoes if you decide to hike through the 
rainforest.
Input current: Even if you're driving through the subtropical rainforest, a few seconds with the doors open while 
you get inside the vehicle is enough time for mosquitoes to get in the vehicle with you.
Output context: Vara, în special, va trebui să ai grijă la țânțari, dacă decizi să faci un traseu prin pădurea 
tropicală.
Output current: Chiar dacă conduci prin pădurea tropicală subtropicală, câteva secunde cu ușile deschise în timp ce
intri în vehicul este suficient timp pentru țânțari să intre în vehicul cu tine.

#1.
Generated output (CTI > 0.121):  Chiar dacă conduci prin pădurea trop(1.747)icală subtropicală, câteva secunde cu 
ușile deschise în timp ce intri în vehicul este suficient timp pentru țânțari să intre în vehicul cu tine.
Input context (CCI > 0.002):     In summer, especially, you'll need to watch out for mosquitoes if you decide to 
hike through the rainforest.
Output context (CCI > 0.002):    Vara, în special, va trebui să ai grijă la țânțari, dacă decizi să faci un traseu 
prin pădurea(0.002) trop(0.004)icală.
#2.
Generated output (CTI > 0.121):  Chiar dacă conduci prin pădurea tropicală subtropicală, câteva secunde cu ușile 
deschise în timp ce intri în vehicul este suficient timp pentru ț(1.416)ânțari să intre în vehicul cu tine.
Input context (CCI > 0.005):     In summer, especially, you'll need to watch out for mosquitoes if you decide to 
hike through the rainforest.
Output context (CCI > 0.005):    Vara, în special, va trebui să ai grijă la ț(0.016)ânțari, dacă decizi să faci un 
traseu prin pădurea tropicală.
#3.
Generated output (CTI > 0.121):  Chiar dacă conduci prin pădurea tropicală sub(0.470)tropicală, câteva secunde cu 
ușile deschise în timp ce intri în vehicul este suficient timp pentru țânțari să intre în vehicul cu tine.
Input context (CCI > 0.003):     In summer, especially, you'll need to watch out for mosquitoes if you decide to 
hike through the rainforest.
Output context (CCI > 0.004):    Vara, în special, va trebui să ai grijă la țânțari, dacă decizi să faci un traseu 
prin pădurea(0.004) trop(0.006)ical(0.004)ă.
#4.
Generated output (CTI > 0.121):  Chiar dacă conduci prin pădurea tropicală subtropicală, câteva secunde cu ușile 
deschise în timp ce intri în vehicul este suficient timp pentru țânțari(0.455) să intre în vehicul cu tine.
Input context (CCI > 0.003):     In summer, especially, you'll need to watch out for mosquitoes if you decide to 
hike through the rainforest.
Output context (CCI > 0.003):    Vara, în special, va trebui să ai grijă la ț(0.006)ân(0.004)țari(0.006), dacă 
decizi să faci un traseu prin pădurea tropicală.
#5.
Generated output (CTI > 0.121):  Chiar(0.307) dacă conduci prin pădurea tropicală subtropicală, câteva secunde cu 
ușile deschise în timp ce intri în vehicul este suficient timp pentru țânțari să intre în vehicul cu tine.
Input context (CCI > 0.008):     In summer, especially, you'll need to watch out for mosquitoes if you decide to 
hike through the rainforest.
Output context (CCI > 0.008):    Vara, în special, va trebui să ai grijă la țânțari, dacă decizi să faci un traseu 
prin pădurea(0.009) tropicală.(0.017)
#6.
Generated output (CTI > 0.121):  Chiar dacă condu(0.268)ci prin pădurea tropicală subtropicală, câteva secunde cu 
ușile deschise în timp ce intri în vehicul este suficient timp pentru țânțari să intre în vehicul cu tine.
Input context (CCI > 0.002):     In summer, especially, you'll need to watch out for mosquitoes if you decide to 
hike through the rainforest.
Output context (CCI > 0.002):    Vara, în special, va trebui să ai grijă la țânțari, dacă(0.002) deci(0.003)zi să 
faci un traseu prin pădurea tropicală.
#7.
Generated out

Sentence 53


Context with contextual cues (std λ=2.00, top 5) followed by output sentence with context-sensitive target spans 
(all)
(CTI = "kl_divergence", CCI = "saliency" w/ "probability" target)

Input context:  Courier companies are well paid for delivering things quickly. Frequently, time is very important 
with business documents, merchandise or spare parts for an urgent repair. On some routes, the larger companies have
their own planes, but for other routes and smaller firms there was a problem. If they sent things by air freight, 
on some routes it may have taken days to get through unloading and customs.
Input current: The only way to get it through faster was to send it as checked luggage. Airline regulations will 
not allow them to send luggage without a passenger, which is where you come in.
Output context: Companiile de curierat sunt bine plătite pentru livrările rapide. Adesea, timpul de livrare este 
esențial în ceea ce privește documentele, mărfurile sau piesele de schimb pentru o reparație urgentă. Pe unele 
rute, companiile mai mari dețin propriile avioane, dar alte rute și companiile mai mici aveau o problemă. Dacă au 
trimis ceva pe calea aerului, pe unele rute poate dura zile întregi descărcarea și vămuirea.
Output current: Singura modalitate de a trece mai repede a fost să o trimiteți ca bagaj înregistrat. Reglementările
companiei aeriene nu le permit să trimită bagaje fără pasager, care este locul în care intri.

#1.
Generated output (CTI > 0.174):  Singura modalitate de a trece mai repede a fost să o trimiteți(2.056) ca bagaj 
înregistrat. Reglementările companiei aeriene nu le permit să trimită bagaje fără pasager, care este locul în care 
intri.
Input context (CCI > 0.001):     Courier companies are well paid for delivering things quickly. Frequently, time is
very important with business documents, merchandise or spare parts for an urgent repair. On some routes, the larger
companies have their own planes, but for other routes and smaller firms there was a problem. If they sent things by
air freight, on some routes it may have taken days to get through unloading and customs.
Output context (CCI > 0.001):    Companiile de curierat sunt bine plătite pentru livrările rapide. Adesea, timpul 
de livrare este esențial în ceea ce privește(0.001) documentele, mărfurile sau piesele de schimb pentru o reparație
urgentă. Pe unele rute, companiile mai mari dețin(0.001) propriile avioane, dar alte rute și companiile mai mici 
aveau o problemă. Dacă au trimis ceva pe calea aerului, pe unele rute poate dura zile între(0.001)gi descărcarea 
și(0.001) vă(0.003)muirea.
#2.
Generated output (CTI > 0.174):  Sing(1.870)ura modalitate de a trece mai repede a fost să o trimiteți ca bagaj 
înregistrat. Reglementările companiei aeriene nu le permit să trimită bagaje fără pasager, care este locul în care 
intri.
Input context (CCI > 0.003):     Courier companies are well paid for delivering things quickly. Frequently, time is
very important with business documents, merchandise or spare parts for an urgent repair. On some routes, the larger
companies have their own planes, but for other routes and smaller firms there was a problem. If they sent things by
air freight, on some routes it may have taken days to get through unloading and customs(0.004).
Output context (CCI > 0.004):    Companiile de curierat sunt bine plătite pentru livrările rapide. Adesea, timpul 
de livrare este esențial în ceea ce privește documentele, mărfurile sau piesele de schimb pentru o reparație 
urgentă. Pe unele rute, companiile mai mari dețin propriile avioane, dar alte rute și companiile mai mici aveau o 
problemă. Dacă au trimis ceva pe calea aerului, pe unele rute poate dura zile întregi descărcarea și 
vă(0.005)muirea(0.006).(0.016)
#3.
Generated output (CTI > 0.174):  Singura modalitate de a trece mai repede a fost să o trimiteți ca bagaj 
înregistrat.(1.067) Reglementările companiei aeriene nu le permit să trimită bagaje fără pasager, care este locul 
în care

Sentence 37


Context with contextual cues (std λ=2.00, top 5) followed by output sentence with context-sensitive target spans 
(all)
(CTI = "kl_divergence", CCI = "saliency" w/ "probability" target)

Input context:  You should choose your Frequent Flyer airline in an alliance carefully.
Input current: Although you may think it is intuitive to join the airline you fly most, you should be aware that 
privileges offered are often different and frequent flyer points may be more generous under a different airline in 
the same alliance.
Output context: Ar trebui să alegeți cu atenție compania aeriană a programului de fidelizare asociată alianței.
Output current: Deși poate credeți că este intuitiv să vă alăturați companiei aeriene pe care le veți zbura cel mai
mult, trebuie să fiți conștienți că privilegiile oferite sunt adesea diferite și că punctele de viteză frecventă 
pot fi mai generoase în cadrul unei alte companii aeriene din aceeași alianță.

#1.
Generated output (CTI > 0.108):  Deși poate credeți că este intuitiv să vă alăturați companiei aeriene pe care le 
veți zbura cel mai mult, trebuie să fiți conștienți că privilegiile oferite sunt adesea diferite și că punctele de 
vite(2.287)ză frecventă pot fi mai generoase în cadrul unei alte companii aeriene din aceeași alianță.
Input context (CCI > 0.003):     You should choose your Frequent Flyer airline in an alliance carefully.
Output context (CCI > 0.003):    Ar trebui să alegeți cu atenție compania aeriană a programului de 
fide(0.008)lizare asociată alianței.
#2.
Generated output (CTI > 0.108):  De(2.011)și poate credeți că este intuitiv să vă alăturați companiei aeriene pe 
care le veți zbura cel mai mult, trebuie să fiți conștienți că privilegiile oferite sunt adesea diferite și că 
punctele de viteză frecventă pot fi mai generoase în cadrul unei alte companii aeriene din aceeași alianță.
Input context (CCI > 0.011):     You should choose your Frequent Flyer airline in an alliance carefully.
Output context (CCI > 0.011):    Ar trebui să alegeți cu atenție compania aeriană a programului de fidelizare 
asociată alianței(0.014).(0.022)
#3.
Generated output (CTI > 0.108):  Deși(1.527) poate credeți că este intuitiv să vă alăturați companiei aeriene pe 
care le veți zbura cel mai mult, trebuie să fiți conștienți că privilegiile oferite sunt adesea diferite și că 
punctele de viteză frecventă pot fi mai generoase în cadrul unei alte companii aeriene din aceeași alianță.
Input context (CCI > 0.001):     You should choose your Frequent Flyer airline in an alliance carefully.
Output context (CCI > 0.001):    Ar trebui să alegeți cu atenție compania aeriană a programului de fidelizare 
asociată alianței(0.001).(0.002)
#4.
Generated output (CTI > 0.108):  Deși poate credeți că este intuitiv să vă alăturați companiei aeriene pe care le 
veți zb(0.548)ura cel mai mult, trebuie să fiți conștienți că privilegiile oferite sunt adesea diferite și că 
punctele de viteză frecventă pot fi mai generoase în cadrul unei alte companii aeriene din aceeași alianță.
Input context (CCI > 0.003):     You should choose your Frequent Flyer airline in an alliance carefully.
Output context (CCI > 0.003):    Ar trebui să alegeți cu atenție compania aeriană a programului de 
fide(0.007)lizare asociată alianței.
#5.
Generated output (CTI > 0.108):  Deși poate credeți că este intuitiv să vă alăturați companiei aeriene pe care le 
ve(0.443)ți zbura cel mai mult, trebuie să fiți conștienți că privilegiile oferite sunt adesea diferite și că 
punctele de viteză frecventă pot fi mai generoase în cadrul unei alte companii aeriene din aceeași alianță.
Input context (CCI > 0.002):     You should choose your Frequent Flyer airline in an alliance carefully.
Output context (CCI > 0.002):    Ar trebui să alegeți cu atenție compania aeriană a programului de 
fide(0.005)lizare asociată alianței.
#6.
Generated output (CTI > 0.108):  Deși poate credeți că este intuitiv să vă alăturați companiei aeriene pe care le 
veți zbura cel mai mu

Sentence 44


Context with contextual cues (std λ=2.00, top 5) followed by output sentence with context-sensitive target spans 
(all)
(CTI = "kl_divergence", CCI = "saliency" w/ "probability" target)

Input context:  Booking in advance gives the traveller peace of mind that they will have somewhere to sleep once 
they arrive at their destination. Travel agents often have deals with specific hotels, although you may find it 
possible to book other forms of accommodation, like camping grounds, through a travel agent. Travel agents usually 
offer packages that include breakfast, transportation arrangements to/from the airport or even combined flight and 
hotel packages.
Input current: They can also hold the reservation for you if you need time to think about the offer or procure 
other documents for your destination (e.g. visa).
Output context: Rezervarea în avans oferă călătorului liniște sufletească că va avea un loc unde să doarmă când va 
ajunge la destinație. Agenții de voiaj au deseori înțelegeri cu anumite hoteluri, dar prin intermediul unui agent 
de voiaj puteți să faceți rezervări și pentru alte forme de cazare, cum ar fi terenuri de camping. Agenții de voiaj
de obicei ofertă pachete care includ micul dejun, transport la și de la aeroport sau chiar pachete care combină 
zborurile și cazarea la hotel.
Output current: Ei pot de asemenea să țină rezervarea pentru tine dacă ai nevoie de timp pentru a te gândi la 
ofertă sau pentru a achiziționa alte documente pentru destinație (de exemplu viză).

#1.
Generated output (CTI > 0.089):  Ei(1.113) pot de asemenea să țină rezervarea pentru tine dacă ai nevoie de timp 
pentru a te gândi la ofertă sau pentru a achiziționa alte documente pentru destinație (de exemplu viză).
Input context (CCI > 0.002):     Booking in advance gives the traveller peace of mind that they will have somewhere
to sleep once they arrive at their destination. Travel agents often have deals with specific hotels, although you 
may find it possible to book other forms of accommodation, like camping grounds, through a travel agent. Travel 
agents usually offer packages that include breakfast, transportation arrangements to/from the airport or even 
combined flight and hotel packages(0.003).(0.003)
Output context (CCI > 0.002):    Rezervarea în avans oferă călătorului liniște sufletească că va avea un loc unde 
să doarmă când va ajunge la destinație. Agenții de voiaj au deseori înțelegeri cu anumite hoteluri, dar prin 
intermediul unui agent de voiaj puteți să faceți rezervări și pentru alte forme de cazare, cum ar fi terenuri de 
camping. Agenții de voiaj de obicei ofert(0.005)ă pachete care includ micul dejun, transport la și de la aeroport 
sau chiar pachete care combină zborurile și cazarea la hotel.(0.008)
#2.
Generated output (CTI > 0.089):  Ei pot de asemenea să țină rezervarea pentru tine dacă ai nevoie de timp pentru a 
te gândi la ofertă sau pentru a achiziționa alte documente pentru destinație(0.598) (de exemplu viză).
Input context (CCI > 0.002):     Booking in advance gives the traveller peace of mind that they will have somewhere
to sleep once they arrive at their destination. Travel agents often have deals with specific hotels, although you 
may find it possible to book other forms of accommodation, like camping grounds, through a travel agent. Travel 
agents usually offer packages that include breakfast, transportation arrangements to/from the airport or even 
combined flight and hotel packages.
Output context (CCI > 0.002):    Rezervarea în avans oferă călătorului liniște sufletească că va avea un loc unde 
să doarmă când va ajunge la destina(0.004)ție(0.006). Agenții de voiaj au deseori înțelegeri cu anumite hoteluri, 
dar prin intermediul unui agent de voiaj puteți să faceți rezervări și pentru alte forme de cazare, cum ar fi 
terenuri de camping. Agenții de voiaj de obicei ofert(0.004)ă pachete care includ micul dejun, transport la și de 
la aeroport sau chiar pachete care combină zborurile și cazarea la h

Sentence 1


Context with contextual cues (std λ=2.00, top 5) followed by output sentence with context-sensitive target spans 
(all)
(CTI = "kl_divergence", CCI = "saliency" w/ "probability" target)

Input context:  How would Io's gravity pull on me? If you stood on the surface of Io, you would weigh less than you
do on Earth.
Input current: A person who weighs 200 pounds (90kg) on Earth would weigh about 36 pounds (16kg) on Io. So the 
gravity, of course, pulls less on you.
Output context: Cum m-ar afecta forța de gravitație pe Io? Dacă ați sta pe suprafața lui Io, ați cântări mai puțin 
decât cântăriți pe Pământ.
Output current: O persoană care cântărește 90 de kilograme pe Pământ ar cânta aproximativ 16 kilograme pe Io. Deci 
gravitația, desigur, vă poate atinge mai puțin.

#1.
Generated output (CTI > 0.180):  O persoană care cântărește(2.183) 90 de kilograme pe Pământ ar cânta aproximativ 
16 kilograme pe Io. Deci gravitația, desigur, vă poate atinge mai puțin.
Input context (CCI > 0.000):     How would Io's gravity pull on me? If you stood on the surface of Io, you would 
weigh less than you do on Earth.
Output context (CCI > 0.000):    Cum m-ar afecta forța de gravitație pe Io? Dacă ați sta pe suprafața lui Io, ați 
cântări mai puțin decât cântăriți pe P(0.000)ământ(0.001).(0.001)
#2.
Generated output (CTI > 0.180):  O persoană care cântărește 90 de kilograme pe Pământ ar cânta aproximativ 16 
kilograme pe Io(1.174). Deci gravitația, desigur, vă poate atinge mai puțin.
Input context (CCI > 0.001):     How would(0.001) Io's gravity pull on me? If you stood on the surface of(0.001) 
Io, you would weigh less than you do on Earth.
Output context (CCI > 0.001):    Cum m-ar afecta forța de gravitație pe Io(0.001)? Dacă ați sta pe suprafața lui 
Io, ați cântări mai puțin decât cântăriți pe Pământ.
#3.
Generated output (CTI > 0.180):  O persoană care cântărește 90 de kilograme pe Pământ ar cânta aproximativ 16 
kilograme pe Io. Deci gravitația, desigur, vă poate(1.089) atinge mai puțin.
Input context (CCI > 0.003):     How would Io's gravity pull on me? If you stood on the surface of Io, you would 
weigh less than you do on Earth.
Output context (CCI > 0.004):    Cum m-ar afecta(0.011) forța de gravitație pe Io? Dacă ați sta pe suprafața lui 
Io, ați cân(0.004)tări mai puțin decât cântăriți pe Pământ.
#4.
Generated output (CTI > 0.180):  O persoană care cântărește 90 de kilograme pe Pământ ar cânta aproximativ 16 
kilograme pe Io.(0.470) Deci gravitația, desigur, vă poate atinge mai puțin.
Input context (CCI > 0.003):     How would Io's gravity pull on me? If you stood on the surface of Io, you would 
weigh less than you do on Earth.
Output context (CCI > 0.004):    Cum m-ar afecta forța de gravitație pe Io? Dacă ați sta pe suprafața lui Io, ați 
cântări mai puțin decât cântăriți pe P(0.008)ământ(0.008).(0.006)
#5.
Generated output (CTI > 0.180):  O(0.408) persoană care cântărește 90 de kilograme pe Pământ ar cânta aproximativ 
16 kilograme pe Io. Deci gravitația, desigur, vă poate atinge mai puțin.
Input context (CCI > 0.035):     How would Io's gravity pull on me? If you stood on the surface of Io, you would 
weigh less than you do on Earth.
Output context (CCI > 0.039):    Cum m-ar afecta forța de gravitație pe Io? Dacă ați sta pe suprafața lui Io, ați 
cântări mai puțin decât cântăriți pe P(0.079)ământ(0.087).(0.063)
#6.
Generated output (CTI > 0.180):  O persoană care cântărește 90 de kilograme pe Pământ ar cânta aproximativ 16 
kilograme pe Io. Deci gravitația, desigur, vă(0.396) poate atinge mai puțin.
Input context (CCI > 0.004):     How would Io's gravity pull on me? If you stood on the surface of Io, you would 
weigh less than you do on Earth.
Output context (CCI > 0.005):    Cum m-ar afecta(0.006) forța de gravitație pe Io? Dacă ați(0.009) sta pe suprafața
lui Io, ați(0.007) cântări mai puțin decât cântăriți pe Pământ.
#7.
Generated output (CTI > 0.180):  O persoană care cântărește 90 de kilograme pe Pământ ar cânta aproximativ 16

Sentence 2


Context with contextual cues (std λ=2.00, top 5) followed by output sentence with context-sensitive target spans 
(all)
(CTI = "kl_divergence", CCI = "saliency" w/ "probability" target)

Input context:  Learning to create interactive media requires conventional and traditional skills, as well as tools
mastered in interactive classes (storyboarding, audio and video editing, story telling, etc.)
Input current: Interactive design requires that you re-assess your assumptions about media production and learn to 
think in a non-linear ways.
Output context: Pentru a învăța să creăm prezentări interactive avem nevoie atât de abilități convenționale și 
tradiționale, cât și de instrumente stăpânite în cursurile interactive (storyboarding, editare audio și video, 
tehnici de narațiune etc.)
Output current: Designul interactiv necesită să vă reevaluati ipotezele despre producția de media și să învățați să
gândiți în mod non-linear.

#1.
Generated output (CTI > 0.056):  Designul interactiv necesită să vă reevaluati ipotezele despre producția de 
media(0.324) și să învățați să gândiți în mod non-linear.
Input context (CCI > 0.002):     Learning to create interactive media requires conventional and traditional skills,
as well as tools mastered in interactive classes (storyboarding, audio and video editing, story telling, etc.)
Output context (CCI > 0.002):    Pentru a învăța să creăm prezent(0.007)ări interactive avem nevoie atât de 
abilități convenționale și tradiționale, cât și de instrumente stăpânite în cursurile interactive (storyboarding, 
editare audio și video, tehnici de narațiune etc.)
#2.
Generated output (CTI > 0.056):  Designul interactiv necesită să vă reevaluati ipotezele despre producția(0.320) de
media și să învățați să gândiți în mod non-linear.
Input context (CCI > 0.001):     Learning to create interactive media requires conventional and traditional skills,
as well as tools mastered in interactive classes (storyboarding, audio and video editing, story telling, etc.)
Output context (CCI > 0.001):    Pentru a învăța să creăm prezentări(0.002) interactive avem nevoie atât de 
abilități convenționale și tradiționale, cât și de instrumente stăpânite în cursurile interactive (storyboarding, 
editare audio și video, tehnici de nara(0.002)ți(0.003)une etc.)
#3.
Generated output (CTI > 0.056):  Design(0.297)ul interactiv necesită să vă reevaluati ipotezele despre producția de
media și să învățați să gândiți în mod non-linear.
Input context (CCI > 0.003):     Learning to create interactive media requires conventional and traditional skills,
as well as tools mastered in interactive classes (storyboarding, audio and video editing, story telling, 
etc(0.003).)(0.003)
Output context (CCI > 0.003):    Pentru(0.004) a învăța să creăm prezent(0.005)ări interactive avem nevoie atât de 
abilități convenționale și tradiționale, cât și de instrumente stăpânite în cursurile interactive (storyboarding, 
editare audio și video, tehnici de narațiune etc.)(0.007)
#4.
Generated output (CTI > 0.056):  Designul interactiv neces(0.151)ită să vă reevaluati ipotezele despre producția de
media și să învățați să gândiți în mod non-linear.
Input context (CCI > 0.002):     Learning to create interactive media requires conventional and traditional skills,
as well as tools mastered in interactive classes (storyboarding, audio and video editing, story telling, etc.)
Output context (CCI > 0.002):    Pentru a învăța să creăm prezentări interactive avem(0.006) nevoie atât de 
abilități convenționale și tradiționale, cât și de instrumente stăpânite în cursurile interactive (storyboarding, 
editare audio și video, tehnici de narațiune etc.)
#5.
Generated output (CTI > 0.056):  Designul interactiv necesită să vă reevaluati ipotezele despre producția de(0.143)
media și să învățați să gândiți în mod non-linear.
Input context (CCI > 0.001):     Learning to create interactive media requires conventional and traditional skills,
as well as tools mastered in interactive 

Sentence 16


Context with contextual cues (std λ=2.00, top 5) followed by output sentence with context-sensitive target spans 
(all)
(CTI = "kl_divergence", CCI = "saliency" w/ "probability" target)

Input context:  The world has over 5,000 different languages, including more than twenty with 50 million or more 
speakers. Written words are often easier to understand than spoken words, also. This is especially true of 
addresses, which are often difficult to pronounce intelligibly.
Input current: Many entire nations are completely fluent in English, and in even more you can expect a limited 
knowledge - especially among younger people.
Output context: În lume există peste 5000 de limbi diferite, inclusiv mai mult de douăzeci cu peste 50 de milioane 
de vorbitori. Cuvântul scris este ades mai ușor de înțeles decât cuvântul vorbit. Acest lucru este cu atât mai 
adevărat în cazul adreselor, care deseori sunt dificil de pronunțat clar.
Output current: Multe națiuni întregi vorbesc complet limba engleză, iar în mai multe cazuri se poate aștepta la o 
cunoaștere limitată - în special printre tineri.

#1.
Generated output (CTI > 0.128):  Multe nați(2.689)uni întregi vorbesc complet limba engleză, iar în mai multe 
cazuri se poate aștepta la o cunoaștere limitată - în special printre tineri.
Input context (CCI > 0.003):     The world has over 5,000 different languages, including more than twenty with 50 
million or more speakers. Written words are often easier to understand than spoken words, also. This is especially 
true of addresses, which are often difficult to pronounce intelligibly.
Output context (CCI > 0.003):    În lume există peste 5000 de limbi diferite, inclusiv mai mult de douăzeci cu 
peste 50 de milioane de vorbitori. Cuvântul scris este ades mai ușor de înț(0.003)eles decât cuvântul vorbit. Acest
lucru este cu atât mai adevărat în cazul adreselor, care deseori sunt dificil de(0.004) pronun(0.007)ț(0.007)at 
clar.
#2.
Generated output (CTI > 0.128):  Mul(0.906)te națiuni întregi vorbesc complet limba engleză, iar în mai multe 
cazuri se poate aștepta la o cunoaștere limitată - în special printre tineri.
Input context (CCI > 0.006):     The world has over 5,000 different languages, including more than twenty with 50 
million or more speakers. Written words are often easier to understand than spoken words, also. This is especially 
true of addresses, which are often difficult to pronounce intelligibly.
Output context (CCI > 0.006):    În lume există peste 5000 de limbi diferite, inclusiv mai mult de douăzeci cu 
peste 50 de milioane de vorbitori. Cuvântul scris este ades mai ușor de înțeles decât cuvântul vorbit. Acest lucru 
este cu atât mai adevărat în cazul adreselor, care deseori sunt dificil de pronunțat clar.(0.021)
#3.
Generated output (CTI > 0.128):  Multe na(0.242)țiuni întregi vorbesc complet limba engleză, iar în mai multe 
cazuri se poate aștepta la o cunoaștere limitată - în special printre tineri.
Input context (CCI > 0.001):     The world has over 5,000 different languages, including more than twenty with 50 
million or more speakers. Written words are often easier to understand than spoken words, also. This is especially 
true of addresses, which are often difficult to pronounce intelligibly(0.001).
Output context (CCI > 0.001):    În lume există peste 5000 de limbi diferite, inclusiv mai mult de douăzeci cu 
peste 50 de milioane de vorbitori. Cuvântul scris este ades mai ușor de înțeles decât cuvântul vorbit. Acest lucru 
este cu atât mai adevărat în cazul adreselor, care deseori sunt dificil de pronun(0.001)țat clar.(0.004)

# **Analysis of the 6 common sentences across all sentences**
**Sentence 12**:  We can observe that the CTI highlighted the pronoun "dumneavoastră" in two instances, one for each sentence of the generated output. In each situation, the almost all highlighted contextual cues (determiners, nouns, word-parts) are not relevant to suggesting a formal pronoun, as generated. This outcome could be explained by the lack of usage of any second person pronouns in the context. However, certain words could imply the usage of a formal versus an informal form, but PECoRe usually fails to detect these tokens, except for the "Afa(0.001)cerilor"(Affairs). Although this token suggests formality, it has a very low score attributed to it, which is in line with the other tokens identified by the CCI step. This fact indicates the model is "coming to the right conclusion for the wrong reasons".

**Sentence 18**: In this example, we can observe a lack of contextual cues, the only ones being represented by the expression "în casă" (wrongly translated) and the conjunction și (and). None of these tokens are contributing to the generation of the context-sensitive target span represented by the verb "călătorești(0.667)" (travel) which implies the informal singular form of you by the suffix "-ști:. The informal form is also maintained for the rest of the sentence, suggested by the verb "așteaptă" (wait). Therefore, similar to the previous case, the model provides a correct translation, but does not rely on context for generating it.

**Sentence 25**: Similar to sentence 18, we can observe a shortage of contextual cues in this case. Moreover, we can observe that the only token highlighted by CTI is the verb "e(0.057)", which is the contracted form of the verb "este" (is), suggesting an informal setting. This is supported by the translation of  "you" to "tine" and the verb form "să ai" (to have). Nevertheless, the model does not manage to justify its decision even though the context contains the verb "a fi" (to be), which is specific to the informal singular form of you.


**Sentence 29**: This instance is the first one in which we can observe that the model rightfully justifies its generated output concerning the phenomenon we are interested in. Two verbs are highlighted by the CTI, "po(1.392)ți" (can) and "vei(0.332)" (will) that indirectly indicate the informal singular form of "you". In each case, the contextual cues are represented by the verbs that suggest the same pronoun form. Although the generated output contains further evidence for the informal singular form of "you" ("te-ai întors"), this is not detected by CTI. This might be due to the fact that it already captured the former instances and this later instance is dependent on them.

**Sentence 39**: Akin to the previous sentence, the informal singular form of you is indicated indirectly through the verb "condu(0.268)ci" (drive), which relies on the contextual cue "deci(0.003)zi" (decide), which indirectly suggests the informal singular form of "you". Other verb forms could have contributed to the generated output but are not captured by either CTI or CCI.

**Sentence 53**: In this case, CTI highlighted the verb "trimiteți(2.056)" (send) that suggests a formal version of the pronoun "you". However, all the highlighted contextual cues are not relevant to suggesting a formal pronoun, as generated. Among these contextual clues is the part-word "vă(0.003)muirea" (customs) which, taken separately as "vă" (you), could suggest the generated output. Regardless if the model interprets vă as you as indicative of the formal tone, it relies on incorrect context to derive its conclusion.


# **Analysis of the remaining 5 sentences**

**Sentence 37**: In this case, the model can justify its generated output concerning the phenomenon we are interested in. Two verbs are highlighted by the CTI, "crede(0.189)ți" (believe) and "ve(0.443)ți" (will) that indirectly indicate the formal singular form of "you". In the former case, the contextual cue is represented by the verb "alege(0.003)ți(0.003)" (choose) that suggests the same pronoun form. Although other contextual cues are highlighted, they are not relevant for generating these verbs, and it might be that due to agreement, the latter verb is dependent on the former verb.


**Sentence 44**: In this case, the model disproves its generated output concerning the phenomenon we are interested in. Two pronouns are highlighted by the CTI,  "te(0.210)" (you) and "tine(0.096)" (you) that indirectly indicate the informal singular form of "you". In each case, the contextual cues are represented by the verbs that suggest the same pronoun form ("pute(0.004)ți(0.004)" (can), "faceți(0.002)" (make)), but formal. Although other contextual cues are highlighted, they are not responsible for generating these pronouns as they include a comma, a preposition, and common nouns.

**Sentence 1**: This translation correctly uses the context "cân(0.004)tări" (weight) and ați(0.007) (you) to generate "poate(1.089)" (can) and "vă(0.396)" (you) markings for formality. Although other contextual cues are highlighted, they are not responsible for generating these.

**Sentence 2**: CTI highlighted the pronoun "vă(0.396)" (you). Nevertheless, no contextual cues that are highlighted are relevant for generating these. This outcome could be explained by the lack of usage of any second-person pronouns in the context.

**Sentence 16**: There is no form of "you", "your", or "yours" present in the output current. The translation (output current) is correct but does not involve the phenomenon of interest. Additionally, an impersonal tone is used during the translation. However, the original translation contains the form "you".